Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewCallout.swift

```diff
@@ -0,0 +1,478 @@
+        $0.setTitleColor(.braveBlurple, for: .normal)
+        $0.backgroundColor = .clear
+        $0.setTitle(" ", for: .normal)
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    }
+    
+    // MARK: - State
+    private(set) var state: WelcomeViewCalloutState?
+    
+    init(pointsUp: Bool) {
+        super.init(frame: .zero)
+        doLayout(pointsUp: pointsUp)
+        
+        [titleLabel, detailsLabel, primaryButton, secondaryButtonContentView].forEach {
+            contentView.addArrangedSubview($0)
+            
+            $0.alpha = 0.0
+            $0.isHidden = true
+        }
+        
+        [primaryButton, secondaryButton].forEach {
+            $0.contentMode = pointsUp ? .bottom : .top
+            $0.snp.makeConstraints {
+                $0.height.equalTo(44.0)
+            }
+        }
+        
+        let leftSpacer = UIView()
+        let rightSpacer = UIView()
+
+        [leftSpacer, secondaryLabel, secondaryButton, rightSpacer].forEach {
+            secondaryButtonContentView.addArrangedSubview($0)
+        }
+        
+        leftSpacer.snp.makeConstraints {
+            $0.width.equalTo(rightSpacer.snp.width)
+        }
+        
+        [titleLabel, detailsLabel].forEach {
+            $0.contentMode = pointsUp ? .bottom : .top
+            
+            $0.snp.makeConstraints {
+                $0.height.greaterThanOrEqualTo(17.0)
+            }
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        layer.shadowColor = #colorLiteral(red: 0.4633028507, green: 0.4875121117, blue: 0.5066562891, alpha: 1).cgColor
+        layer.shadowOpacity = 0.36
+        layer.shadowOffset = CGSize(width: 5, height: 5)
+        layer.shadowRadius = DesignUX.cornerRadius
+    }
+    
+    private func doLayout(pointsUp: Bool) {
+        arrowView.removeFromSuperview()
+        contentView.removeFromSuperview()
+        
+        if pointsUp {
+            addSubview(backgroundView)
+            addSubview(arrowView)
+            addSubview(contentView)
+            arrowView.transform = .identity
+            
+            arrowView.snp.makeConstraints {
+                $0.centerX.equalToSuperview()
+                $0.top.equalToSuperview()
+                $0.width.equalTo(20.0)
+                $0.height.equalTo(13.0)
+            }
+            
+            contentView.snp.makeConstraints {
+                $0.top.equalTo(arrowView.snp.bottom)
+                if UIDevice.isIpad {
+                    $0.width.lessThanOrEqualTo(450)
+                    $0.centerX.equalToSuperview()
+                } else {
+                    $0.leading.trailing.equalToSuperview().inset(DesignUX.padding)
+                }
```

Comment:
Use size classes instead, an iPad can still be laid out with iPhone layout with multitasking layouts

Reply:
Fixed at https://github.com/brave/brave-ios/pull/4508/commits/b44eeec35605eb23f6d11a8aeee16692fdb859c2